crates/tests/test-tedge-cli/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+mod test_env;
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+    use assert_matches::*;
+    use rexpect::errors::*;
+    use rexpect::process::signal::Signal;
+    use rexpect::process::wait::WaitStatus;
+    use rexpect::*;
+
+    macro_rules! execute_cmd {
```

Comment:
I think that we can really improve test clarity with macros.

```
execute_cmd!("tedge config set c8y.url {url}");
```

Reply:
How about renaming it to 'spawn_and_wait' or somesuch to make it clear that this macro will block